internal/app/singularity/verify_ocsp.go

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) 2022, Sylabs Inc. All rights reserved.
+// Copyright (c) 2020-202, ICS-FORTH.  All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the LICENSE.md file
+// distributed with the sources of this project regarding your rights to use or distribute this
+// software.
+
+package singularity
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/x509"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/sylabs/singularity/pkg/sylog"
+	"golang.org/x/crypto/ocsp"
+)
+
+/*
+Online Certificate Status Protocol - OCSP
+
+OCSP responder is used to provide real-time verification of the revocation status of an X.509 certificate.
+RFC: https://www.rfc-editor.org/rfc/rfc6960
+*/
+
+const (
+	PKIXOCSPNoCheck = "1.3.6.1.5.5.7.48.1.5"
```

Comment:
Would be nice to have a bit more of a note what this is... for anyone who hasn't grepped the OCSP docs.